### DISCUSSION TITLE
How can I filter by a value inside a pivot table?

### DISCUSSION THREAD

**Discussion by lumenwrites** [prisma/prisma#13239] [Q&A]
Hi! I'm building a bike rental app where users can reserve bikes for a certain number of days. Bikes and users have a many-to-many relationship, there's a bike_user pivot table that contains the information about reservation start and end dates. See the diagram:
![bikerr](https://user-images.githubusercontent.com/4761235/167254623-e74cc0ce-7cd7-4487-a0a0-a3ef499b6c41.png)

My schema looks kinda like this:

```
model User {
  id            String @id @default(cuid())
  username      String @unique
  email         String @unique
  rented_bikes  BikeUser[]
}


model Bike {
  id              String @id @default(cuid())
  model           String
  color           String
  rented_by_users BikeUser[]
}

model BikeUser {
  user                      User @relation(fields: [userId], references: [id])
  userId                    String
  bike                      Bike @relation(fields: [bikeId], references: [id])
  bikeId                    String
  reservation_start_date    DateTime 
  reservation_end_date      String

  @@id([userId, bikeId])
}
```

My goal is to take a range of dates, and filter the bikes, showing only the ones that haven't been reserved between these two dates.

Can you please help me figure out how to do this?

**Comment by austincrim** [comment]
Hey @lumenwrites 👋,

When is the `BikeUser` record created? What does the data look like if the same user tries to reserve the same bike for two different time periods?

Just trying to fully understand the data model to help me come up with a solution.

**Reply by vladi-stevanovic** [reply]
👋 Hello @lumenwrites  I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! 🙌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
